spec/repositories/validated_asset_repository_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe ValidatedAssetRepository do
+  subject { ValidatedAssetRepository.new(Image) }
+
+  describe "#new" do
+    let(:image) { Image.new }
+    before do
+      allow(Image).to receive(:new).and_return(image)
+      allow(image).to receive(:create_or_update).and_return(true)
+    end
+    let(:result) { subject.new }
+    it "should validate it" do
+      result.lcsubject = ["bla"]
+
+      expect(result).not_to be_valid
+    end
+    it "should not be able to save it with a bad subject" do
+      result.lcsubject = ["bla"]
```

Comment:
Would it make sense to throw in a test showing that a string that *looks* like a URI is still invalid?